src/lib/y2storage/fake_device_factory.rb

```diff
@@ -413,6 +437,34 @@ def create_free(parent, args)
       disk_name
     end
 
+    # Factory method to create an encryption layer.
+    #
+    # @param parent [String] parent device name ("/dev/sda1" etc.)
+    #
+    # @param args [Hash] encryption layer parameters:
+    #   "name"     name encryption layer ("cr_Something")
+    #   "type"     encryption type; default: "luks"
+    #   "password" encryption password (optional)
+    #
+    # @return [Object] new encryption object
+    #
+    def create_encryption(parent, args)
+      log.info("#{__method__}( #{parent}, #{args} )")
+      name = args["name"]
+      raise ArgumentError, "\"name\" missing for encryption on #{parent}" if name.nil?
+      password = args["password"]
+      type_name = args["type"] || "luks"
+      # We only support creating LUKS so far
+      raise ArgumentError, "Unsupported encryption type #{type_name}" unless type_name == "luks"
+      encryption = ::Storage::Luks.create(@devicegraph, name)
+      encryption.password = password unless password.nil?
+      if @partitions.key?(parent)
+        # Notify create_file_system that this partition is encrypted
+        @partitions[parent]["encryption"] = encryption.name
+      end
+      encryption
```

Comment:
It's not clear to me where the parent and the encryption are linked together. For example, if you do `a_partition.create_encryption(name)` you can then do `a_partition.encryption`. So it's pretty clear than the partition is encrypted. Here, we create the Luks device by and and I guess that link is done in another step, but I'm not sure where/when.